rpc/glfs-operations.c

```diff
@@ -674,3 +677,138 @@ blockGetMetaInfo(struct glfs* glfs, char* metafile, MetaInfo *info,
 
   return ret;
 }
+
+
+int
+blockGetPrioPathIndex(struct glfs* glfs, char *volume, blockServerDefPtr list)
+{
+  struct glfs_fd *pfd = NULL;
+  char attr[256 + 12];
+  char buf[1024];
+  int index = 0;
+  size_t count = 0;
+  size_t min = 0;
+  bool flag = true;
+  int i;
+  int ret = -1;
+
+
+  pfd = glfs_creat(glfs, GB_PRIO_FILE, O_RDONLY | O_CREAT | O_SYNC,
+                   S_IRUSR | S_IWUSR);
+  if (!pfd) {
+    LOG("gfapi", GB_LOG_ERROR, "glfs_creat(%s) on volume %s failed[%s]",
+        GB_PRIO_FILE, volume, strerror(errno));
+    return 1;
+  }
+
+  for (i = 0; i < list->nhosts; i++) {
+    memset(attr, '\0', 256 + 12);
+    snprintf(attr, 256 + 12, "user.block.%s", list->hosts[i]);
+
+    memset(buf, '\0', 1024);
+    if (glfs_fgetxattr(pfd, attr, buf, 1024) < 0) {
+      if (errno != ENODATA) {
+        LOG("gfapi", GB_LOG_ERROR,
+            "glfs_fgetxattr(%s) on volume %s for prio file %s failed[%s]",
+            attr, volume, GB_PRIO_FILE, strerror(errno));
+        goto out;
+      } else {
+        index = i;
+        min = 0;
+        break;
+      }
+    }
+
+    sscanf(buf, "%zu", &count);
+    if (flag || min > count) {
+      min = count;
+      index = i;
+      flag = false;
+    }
+  }
+
+  ret = 0;
+
+ out:
+  if (pfd && glfs_close(pfd) != 0) {
+    LOG("gfapi", GB_LOG_ERROR, "glfs_close(%s): on volume %s failed[%s]",
+        GB_PRIO_FILE, volume, strerror(errno));
+  }
+
+  if (ret)
+    return 1;        /* on failure let the prio index be 1 */
+
+  return index + 1;
+}
+
+
+void
+blockIncPrioAttr(struct glfs* glfs, char *volume, char *addr)
+{
+  size_t count;
+  char buf[1024] = {'\0', };
+  char attr[256 + 12] = {'\0', };
```

Comment:
xattr key length has a limit of 255 chars. Value has a limit of 64KB. http://man7.org/linux/man-pages/man5/attr.5.html So this needs to change.